streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamsPartitionAssignor.java

```diff
@@ -638,8 +639,24 @@ public Subscription subscription(final Set<String> topics) {
         // ---------------- Step Three ---------------- //
 
         // construct the global partition assignment per host map
+        final Map<HostInfo, Set<TaskId>> tasksByHostState = new HashMap<>();
         final Map<HostInfo, Set<TopicPartition>> partitionsByHostState = new HashMap<>();
-        if (minReceivedMetadataVersion >= 2) {
+        if (minReceivedMetadataVersion >= 4) {
+            for (final Map.Entry<UUID, ClientMetadata> entry : clientMetadataMap.entrySet()) {
+                final HostInfo hostInfo = entry.getValue().hostInfo;
+
+                if (hostInfo != null) {
+                    final Set<TaskId> taskIds = new HashSet<>();
+                    final ClientState state = entry.getValue().state;
+
+                    for (final TaskId id : state.activeTasks()) {
+                        taskIds.add(id);
+                    }
+
+                    tasksByHostState.put(hostInfo, taskIds);
+                }
+            }
+        } else {
             for (final Map.Entry<UUID, ClientMetadata> entry : clientMetadataMap.entrySet()) {
```

Comment:
It seems this still needs to be executed for `minReceivedMetadataVersion >= 2`

Reply:
Done.